### ISSUE TITLE
Sending too many parameters to an eth_X function crashes testrpc

### ISSUE THREAD

**Issue by rmeissner** [ConsenSys-archive/ganache-cli-archive#431]
## Expected Behavior
When calling an eth_X function with a wrong set of parameters a 400 should be returned (or at least the testrpc should not crash)

## Current Behavior
if too many parameters are send with an eth_X call than the client crashes with the following error:

```
/usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:74143
var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object"&&"function"==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;ENVIRONMENT_IS_

TypeError: callback is not a function
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:90468:5
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:22449:5
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:22433:7
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67501:5
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67867:14
    at Object.return (/usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67886:9)
    at processNode (/usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67709:30)
    at processNode (/usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67914:5)
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67910:11
    at /usr/local/lib/node_modules/ethereumjs-testrpc/build/cli.node.js:67604:7
```

## Steps to Reproduce (for bugs)

1. start testrpc (tested with `testrpc -s 42`)
2. send `eth_getBalance` (was also tested with `eth_getTransactionReceipt`) with an additional parameter (e.g. `false`)
```
curl -X POST \
  http://localhost:8545/ \
  -H 'content-type: application/json' \
  -d '{
	"method": "eth_getBalance",
	"params": ["0xa5056c8efadb5d6a1a6eb0176615692b6e648313", "latest", false],
	"jsonrpc":"2.0",
	"id": 45
}'
```

## Context
I was sending request from a client and accidentally build the wrong parameter array. As the error was super confusing it took a while to discover the source.

## Your Environment
* Version used: EthereumJS TestRPC v6.0.3 (ganache-core: 2.0.2)
* Software: curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.11 nghttp2/1.24.0
* Operating System and version: Mac High Sierra
* Link to your project: https://github.com/gnosis/heimdall-android

**Comment by benjamincburns** [comment]
Thanks for reporting this @rmeissner - I'm sorry it's gone unnoticed for so long!

I ran into this today, myself. I'll make sure it's fixed in time for the 7.0 release.

Note to self or anyone who wants to take this on as a PR: to fix, add a check for `request.params.length == method.length - 1` in geth_api_double's `handleAsync`, plus relevant tests.

**Comment by benjamincburns** [comment]
Turns out that this case duplicates #255. Since that's the older issue, lets track this over there.
